### ISSUE TITLE
Allow `web+ap://` as alias for `https://` in search

### ISSUE THREAD

**Issue by SoniEx2** [syuilo/Misskey#9504]
## Summary

As stated: `web+ap://` should be just as acceptable as `https://`. Also, `web+ap://` should behave the same as `https://`.

(Do *not*, however, call registerProtocolHandler. The UX for it is a nightmare, there's a better approach being worked on.)

**Comment by saschanaz** [comment]
>The UX for it is a nightmare, there's a better approach being worked on.

Is there some reference for this argument, and also for `web+ap`?

**Comment by SoniEx2** [comment]
yes, mastodon dropped support for registerProtocolHandler because it didn't align with mastodon's design goals. tho it didn't use `web+ap` back then, we don't remember what it used. see https://github.com/mastodon/mastodon/issues/19679

`web+ap` is our attempt to make "web-based activitypub URLs". we've been working on this stuff over at https://github.com/fedi-to/fedi-to.github.io, tho the "workaround", https://github.com/fedi-to/activitypub-helper, didn't seem to work with misskey out of the box last we checked (we might have checked wrong, however). but for mastodon, well, https://fedi-to.net/go?h=3&target=web%2Bap://chaos.social/@SoniEx2. the other relevant part is https://github.com/fedi-to/fedi-to.github.io/blob/main/protocol-handler.md but we're still waiting for IANA approval on that one (not that we can't *use* it, but it'd be a bit rude).

**Comment by saschanaz** [comment]
> `web+ap` is our attempt to make "web-based activitypub URLs". we've been working on this stuff over at https://github.com/fedi-to/fedi-to.github.io, tho the "workaround",

I'm not sure who are "we" here, are you trying to gather some interest or are there some actual discussion among implementors about this?

**Comment by SoniEx2** [comment]
trying to gather some interest, yeah.

**Comment by SoniEx2** [comment]
(for what it's worth, the /.well-known/protocol-handler stuff is already live. no fediverse instance supports it yet tho, and IANA hasn't given an okay yet either. but anyway.)

**Comment by saschanaz** [comment]
> (for what it's worth, the /.well-known/protocol-handler stuff is already live. no fediverse instance supports it yet tho, and IANA hasn't given an okay yet either. but anyway.)

What does it mean by "already live" in that case?

**Comment by SoniEx2** [comment]
e.g. https://fedi-to.net/go?target=web%2Bganarchy://ganarchy.github.io/cc8ab1441fd398d23454c149c74bd5a2304eb566

(check network log in developer tools when visiting this)

**Comment by saschanaz** [comment]
Well, so being live means you made something with your own proposal and there's no one else interested. In that case I don't think there is any kind of consensus that your suggestion is ideal for the fediverse, and thus this should be a discussion rather than a feature request.

**Comment by saschanaz** [comment]
(And such discussion should happen in a standardization group e.g. https://github.com/w3c/activitypub instead of here)

**Comment by SoniEx2** [comment]
we've been trying to get standardization groups to take a look at this but what we've found is that they're deadlocked on eachother.

registerProtocolHandler is confusing so activitypub/mastodon/etc won't add a protocol handler, and nobody uses protocol handlers so browsers won't improve the UX and make it less confusing.

so then do we just abandon interop entirely and rely on mastodon-specific browser extensions that don't work with misskey? or do we just make fedi-to, and then try to get projects to support it as a third-party service?

worst case: we fail. but it'd be nice to be able to link to our mastodon from our blog, and have it open in the user's fediverse app if the user has one registered, but otherwise just open it in our instance. this latter bit is the main thing we're going for here.

**Comment by saschanaz** [comment]
>we've been trying to get standardization groups to take a look at this but what we've found is that they're deadlocked on eachother.

Could you provide a reference for this argument?

**Comment by SoniEx2** [comment]
we tried to get vivaldi to tweak protocol handlers for their browser https://forum.vivaldi.net/topic/82106/revamp-web-protocol-handlers-for-improved-fediverse-integration

we also looked into asking firefox about it but we can't remember if we ever figured out where to ask about it.

and further, y'know, mastodon removed support for them due to the UX stuff. https://github.com/mastodon/mastodon/issues/19679

**Comment by saschanaz** [comment]
Oh, I just found your bug (that sadly nobody responded to): https://bugzilla.mozilla.org/show_bug.cgi?id=1712014
